src/fondant/data_io.py

```diff
@@ -137,7 +141,7 @@ def load_dataframe(self) -> dd.DataFrame:
                 subset_df,
                 left_index=True,
                 right_index=True,
-                how="inner",
+                how="left",
```

Comment:
Don't know why this was "inner" before. "left" is what we want to do (filter on the index).

Reply:
The reason for that was due to an error that Niels was facing which got fixed when changing the join type. I'll link you to the chat. 

I am not a DE expert but I read that inner joins should generally be faster and since an inner join only joins matching ids it should also filter on the index.

Reply:
I think left join is what we want functionally. The index keeps track of which records are in the dataset, so we should use its ids. If data for a certain id is missing in a subset, that will become None instead, which I think is quite logical.

I think the issue Niels had before might be solved by the changes in this PR as well, let's try.

As far as I know, inner joins are only more performant because of their different functionality. If all ids are present in the right index (which should usually be the case), I believe there is no difference. If ids are missing in the right index, we're not getting the functionality we want.

Reply:
> I If data for a certain id is missing in a subset, that will become None instead, which I think is quite logical.

Won't this require dropping all the Nons as well before passing the dataframe to avoid having to handle cases where the row does not contain the expected data? Otherwise the user transform function will have to handle this.

Reply:
I think it's better not to drop it, because it would be silent and hard to debug. Instead it now fails loudly, or the user code in the component needs to handle Nones.